### PR TITLE
Use C#8

### PR DESCRIPTION
**Pull request by benaadams**
Testing

### CODE REVIEW

**Comment by benaadams** [comment]
```
CSC : error CS1617: Invalid option '8.0' for /langversion. 
Use '/langversion:?' to list supported values.
 [D:\j\workspace\x64_release_w---8f801950\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

**Comment by jkotas** [comment]
This would need to be updated https://github.com/dotnet/buildtools and/or https://github.com/dotnet/arcade first.

File: src/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 1]
```diff
@@ -35,7 +35,7 @@
     <_TargetFrameworkDirectories>$(MSBuildThisFileDirectory)/Documentation</_TargetFrameworkDirectories>
     <_FullFrameworkReferenceAssemblyPaths>$(MSBuildThisFileDirectory)/Documentation</_FullFrameworkReferenceAssemblyPaths>
     <SkipCommonResourcesIncludes>true</SkipCommonResourcesIncludes>
-    <LangVersion>latest</LangVersion>
+    <LangVersion>8.0</LangVersion>
```
Comment (jkotas) [inline src/System.Private.CoreLib/System.Private.CoreLib.csproj]: latest would give you 8.0 anyway...

Comment (benaadams) [reply src/System.Private.CoreLib/System.Private.CoreLib.csproj]: Doesn't currently; complains 7.3 doesn't support the static local functions and to upgrade to 8.0 (in VS) when using latest.

Guess its "`latest` released"

**Comment by benaadams** [comment]
Raised issues https://github.com/dotnet/buildtools/issues/2219 and https://github.com/dotnet/arcade/issues/1903

File: src/System.Private.CoreLib/System.Private.CoreLib.csproj (hunk 1 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/System.Private.CoreLib.csproj]: I believe `latest` is still coded for 7.3 and you have to explicitly opt into 8.0: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/LanguageVersion.cs,480397a440092f0c,references

CC. @jaredpar

Comment (jkotas) [reply src/System.Private.CoreLib/System.Private.CoreLib.csproj]: thanks for the pointer

**Comment by tarekgh** [comment]
should we hold a little more using C# 8.0 as there still some features that compiler depend on still undergoing? we should pick the updated bits when it is ready. 

CC @agocke

**Comment by tarekgh** [comment]
@tannergooding I am not sure I was clear enough. We already have open PR to do Index/Range updates. I don't know if this get merged and in same time using the compiler bits that depending on the old Index/Range types will cause problems. in other word, I am afraid we'll run into situation the compiler and the types are not in sync and the result can be undesirable.
I am already CC'ed @agocke to advise about that.

**Comment by stephentoub** [comment]
> I don't know if this get merged and in same time using the compiler bits that depending on the old Index/Range types will cause problems.

It should only matter if we actually use the language feature, and in the case of index/range, I generally don't think we should until the perf issues are addressed.

**Comment by agocke** [comment]
It's likely that the updates to the API will break support for Index/Range syntax, but there's nothing to be done about that. I agree with @stephentoub that you can just avoid using those features right now. Using the types should still be fine.

**Comment by stephentoub** [comment]
Closing.  Can't update until we get updated compilers in the repos.

**Pull request closed without merging**